pl_bolts/models/self_supervised/simclr/simclr_module.py

```diff
@@ -74,197 +87,371 @@ def __init__(self,
+        parser.add_argument("--start_lr", default=0, type=float, help="initial warmup learning rate")
+        parser.add_argument("--final_lr", type=float, default=1e-6, help="final learning rate")
 
         return parser
 
 
 def cli_main():
     from pl_bolts.callbacks.ssl_online import SSLOnlineEvaluator
-    from pl_bolts.datamodules import CIFAR10DataModule, ImagenetDataModule, STL10DataModule
+    from pl_bolts.models.self_supervised.simclr.transforms import SimCLREvalDataTransform, SimCLRTrainDataTransform
+    from pl_bolts.datamodules import STL10DataModule, CIFAR10DataModule, ImagenetDataModule
 
     parser = ArgumentParser()
 
-    # trainer args
-    parser = pl.Trainer.add_argparse_args(parser)
-
     # model args
     parser = SimCLR.add_model_specific_args(parser)
     args = parser.parse_args()
 
-    # init default datamodule
-    if args.dataset == 'cifar10':
-        dm = CIFAR10DataModule.from_argparse_args(args)
-        dm.train_transforms = SimCLRTrainDataTransform(32)
-        dm.val_transforms = SimCLREvalDataTransform(32)
-        args.num_samples = dm.num_samples
+    if args.dataset == 'stl10':
+        dm = STL10DataModule(
+            data_dir=args.data_path,
+            batch_size=args.batch_size,
+            num_workers=args.num_workers
+        )
 
-    elif args.dataset == 'stl10':
-        dm = STL10DataModule.from_argparse_args(args)
         dm.train_dataloader = dm.train_dataloader_mixed
         dm.val_dataloader = dm.val_dataloader_mixed
         args.num_samples = dm.num_unlabeled_samples
 
-        (c, h, w) = dm.size()
-        dm.train_transforms = SimCLRTrainDataTransform(h)
-        dm.val_transforms = SimCLREvalDataTransform(h)
+        args.maxpool1 = False
+        args.first_conv = True
+        args.input_height = dm.size()[-1]
+
+        normalization = stl10_normalization()
 
-    elif args.dataset == 'imagenet2012':
-        dm = ImagenetDataModule.from_argparse_args(args, image_size=196)
-        (c, h, w) = dm.size()
-        dm.train_transforms = SimCLRTrainDataTransform(h)
-        dm.val_transforms = SimCLREvalDataTransform(h)
+        args.gaussian_blur = True
+        args.jitter_strength = 1.
+    elif args.dataset == 'cifar10':
+        val_split = 5000
+        if args.nodes * args.gpus * args.batch_size > val_split:
+            val_split = args.nodes * args.gpus * args.batch_size
+
+        dm = CIFAR10DataModule(
+            data_dir=args.data_path,
+            batch_size=args.batch_size,
+            num_workers=args.num_workers,
+            val_split=val_split
+        )
+
+        args.num_samples = dm.num_samples
+
+        args.maxpool1 = False
+        args.first_conv = False
+        args.input_height = dm.size()[-1]
+        args.temperature = 0.5
+
+        normalization = cifar10_normalization()
+
+        args.gaussian_blur = False
+        args.jitter_strength = 0.5
```

Comment:
this hardcoded change will be confusing for the user as he asks something and gets something else, at least this has to be described in docs

Reply:
these are standard for the dataset. I'll update the docs to mention this but these the params you use to train with this dataset.